cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -269,11 +318,20 @@ private void clearOutputs() {
   }
 
   /**
+   * Gets the MapperInputs for this MapReduce job.
+   *
+   * @return the MapperInputs for the MapReduce job
+   */
+  Map<String, MapperInput> getMapperInputs() {
+    return ImmutableMap.copyOf(inputs);
+  }
```

Comment:
Currently, we don't need to copy the `inputs` map, because we know that `setInput` or `addInput` will not be called after our current usage of `getMapperInputs`, but I don't think it hurts to copy it either, in case that changes in the future.